test/technology/sidb_lattice.cpp

```diff
@@ -0,0 +1,52 @@
+//
+// Created by Jan Drewniok on 12.02.24.
+//
+
+#include "catch2/catch_test_macros.hpp"
+
+#include <fiction/layouts/cell_level_layout.hpp>
+#include <fiction/technology/sidb_lattice.hpp>
+#include <fiction/technology/sidb_lattice_orientations.hpp>
+#include <fiction/traits.hpp>
+#include <fiction/types.hpp>
+
+using namespace fiction;
+
+TEST_CASE("layout traits", "[sidb-lattice]")
+{
+    CHECK(!is_sidb_lattice_v<sidb_cell_clk_lyt>);
+    CHECK(is_cell_level_layout_v<sidb_cell_clk_lyt>);
+    CHECK(!is_sidb_lattice_111_v<sidb_cell_clk_lyt>);
+    CHECK(!is_sidb_lattice_100_v<sidb_cell_clk_lyt>);
+
+    using lattice_layout_100 = sidb_lattice<sidb_100_lattice, sidb_cell_clk_lyt>;
+    CHECK(is_sidb_lattice_v<lattice_layout_100>);
+    CHECK(is_cell_level_layout_v<lattice_layout_100>);
+    CHECK(is_sidb_lattice_100_v<lattice_layout_100>);
+
+    using lattice_layout_111 = sidb_lattice<sidb_111_lattice, sidb_cell_clk_lyt>;
+    CHECK(is_sidb_lattice_v<lattice_layout_111>);
+    CHECK(is_cell_level_layout_v<lattice_layout_111>);
+    CHECK(is_sidb_lattice_111_v<lattice_layout_111>);
+
+    using lattice_on_lattice = sidb_lattice<sidb_100_lattice, lattice_layout_111>;
+    CHECK(is_sidb_lattice_v<lattice_on_lattice>);
+    CHECK(is_cell_level_layout_v<lattice_on_lattice>);
+    CHECK(is_sidb_lattice_111_v<lattice_on_lattice>);
```

Comment:
This should not compile in my opinion. It is a design decision obviously. Maybe we can ponder this question together. If you have a 111 surface already and you try to layer a 100 on top, that might indicate a problem with your code. That's my reasoning behind why I think it should not compile.